modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,736 @@
+	addtimer(CALLBACK(src, .proc/finish_deploy), SETUP_TIME)
+	flick("deploying", src)
+
+/obj/structure/bed/borg_action_pacifier/proc/finish_deploy()
+	name = "cyborg control-bay"
+	icon_state = "up"
+	deployed = TRUE
+
+	if(obj_flags & EMAGGED)
+		// MEATBAG TERMINATION MODE ONLINE - line 478~
+		var/mob/living/simple_animal/hostile/borg_action_pacifier/BAPer = new (get_turf(src))
+		BAPer.power_storage = power_storage // Gotta keep that boom
+		BAPer.check_smoke() // Telegraph how bigly we explode on death
+		BAPer.balloon_alert_to_viewers("ERROR!")
+		BAPer.emote("exclaim")
+		qdel(src)
+	else
+		balloon_alert_to_viewers("deployed")
+		playsound(src, 'sound/machines/ping.ogg', 25, FALSE, falloff_exponent = 10)
+
+/obj/structure/bed/borg_action_pacifier/MouseDrop(over_object, src_location, over_location)
+	. = ..()
+	if((over_object == usr) && Adjacent(usr))
+		if(!ishuman(usr) || !usr.canUseTopic(src, BE_CLOSE))
+			return FALSE
+		if(has_buckled_mobs() || deployed)
+			return FALSE
+
+		usr.visible_message(span_notice("[usr] shuts off [src]."), span_notice("You shut off [src]."))
+		var/obj/item/grenade/borg_action_pacifier_grenade/BAPer = new (get_turf(src))
+		usr.put_in_hands(BAPer)
+		BAPer.power_storage = power_storage
+		qdel(src)
+
+// Radial menu for the deployed model's function controls
+/obj/structure/bed/borg_action_pacifier/proc/check_alt_clicked_radial(datum/source, mob/living/clicker)
+	SIGNAL_HANDLER
+
+	if(!istype(clicker))
+		return
+	. = COMPONENT_CANCEL_CLICK_ALT
+	INVOKE_ASYNC(src, .proc/alt_clicked_radial, clicker)
+
+/obj/structure/bed/borg_action_pacifier/proc/alt_clicked_radial(mob/living/clicker)
+	var/icon/radial_indicator = 'modular_skyrat/modules/deployables/icons/deployable_indicator.dmi'
+	var/list/choices = list()
+
+	if(buckled_cyborg)
+		if(!locked)
+			choices += list(LOCK = image(icon = radial_indicator, icon_state = "lock"))
+		else
+			choices += list(UNLOCK = image(icon = radial_indicator, icon_state = "unlock"))
+
+		switch(enabled_function)
+			if(NONE)
+				if(buckled_cyborg.cell && buckled_cyborg.cell.charge > 0)
+					choices += list(DRAIN_MODE = image(icon = radial_indicator, icon_state = "drain"))
+				if(buckled_cyborg.cell && power_storage > 0)
+					choices += list(PUMP_MODE = image(icon = radial_indicator, icon_state = "pump"))
+			else
+				choices += list(STOP_MODE = image(icon = radial_indicator, icon_state = "stop"))
+	else
+		set_mode(clicker, FOLD)
+		return
+
+	var/choice = show_radial_menu(clicker, src, choices, custom_check = CALLBACK(src, .proc/check_menu, clicker), tooltips = TRUE)
+	if(!choice || !check_menu(clicker))
+		return
+	set_mode(clicker, choice)
+
+/obj/structure/bed/borg_action_pacifier/proc/set_mode(mob/living/clicker, choice)
+	switch(choice)
+		if(LOCK)
+			balloon_alert_to_viewers("locked")
+			lock(clicker)
+		if(UNLOCK)
+			balloon_alert_to_viewers("unlocked")
+			unlock(clicker)
```

Comment:
This bypasses Robotics access?

Reply:
yes! this is intended

Reply:
☹️

Reply:
i could add a check that requires the panel to be open?